src/components/assignment-panel/assignment-info/assignment-info.tsx

```diff
@@ -212,6 +231,60 @@ export const AssignmentInfo = ({  }: AssignmentInfoProps) => {
                     />
                 </div>
             </div>
+            <div className={ assignmentInfoSectionClass } style={{ marginTop: 0 }}>
+                <h5 className={ assignmentInfoSectionHeaderClass }>
+                    Master notebook
+                    { gradedNotebookInvalid && ` (invalid)` }
+                    <InfoTooltip
+                        title="This notebook contains Otter Grader test cases"
+                        trigger="hover"
+                        iconProps={{ style: { fontSize: 13, marginLeft: 4 } }}
+                    />
+                </h5>
+                <div style={{ width: "100%" }}>
+                    <Select
+                        error={ gradedNotebookInvalid }
+                        defaultValue={ assignment.masterNotebookPath }
+                        onChange={ (e: ChangeEvent<any>) => {
+                            setGradedNotebookControlled(e.target.value)
+                            onGradedNotebookChanged(e)
+                        } }
+                        style={{ width: "100%" }}
+                    >
+                        {
+                            notebookFiles[assignment.id].map((notebook) => (
+                                <MenuItem key={ notebook } value={ notebook }>{ notebook }</MenuItem>
+                            ))
+                        }
+                    </Select>
+                    { !gradedNotebookInvalid && (
+                        <FormHelperText style={{ color: "#1976d2" }}>
+                            <a onClick={ openGradedNotebook } style={{ cursor: "pointer" }}>
+                                Open notebook
+                            </a>
+                        </FormHelperText>
+                    )}
+                </div>
+            </div>
+            {/* <div className={ assignmentInfoSectionClass } style={{ marginTop: 0 }}>
```

Comment:
Do we need this commented code?

Reply:
In the event we choose to add it

Reply:
I think if you ever keep commented code in the repo you should add a comment explaining the scenario in which it should be uncommented. Otherwise someone else might remove later it thinking it's not needed code. If you agree can you add that comment above this block?

Reply:
I agree, will add